scripts/core/editor3/components/Editor3.jsx

```diff
@@ -20,146 +21,137 @@ import Toolbar from './toolbar';
  * @description Editor3 is a draft.js based editor that support customizable
  *  formatting (shortly more features will come ...)
  */
-export class Editor3 extends React.Component {
-    constructor(props) {
-        super(props);
 
-        const initialContentState = stateFromHTML(props.value);
+export const getInitialState = (spellcheckerService, value, onChangeValue, readOnly, showToolbar, editorFormat) => {
+    const initialContentState = stateFromHTML(value);
 
-        this.state = {
-            editorState: EditorState.createWithContent(
+    return {
+        editorState: EditorState.createWithContent(
                 initialContentState,
-                this.getDecorators()
-            )
-        };
+                getDecorators(spellcheckerService)
+        ),
+        spellcheckerState: {spellcheckerService: spellcheckerService},
```

Comment:
I think we should remove this key (spellcheckerState) and instead just pass the Angular `$injector` which will be used by other components too. It will allow obtaining any service, including the spellchecker service.